src/psyclone/tests/psyir/nodes/dynamic_omp_task_directive_test.py

```diff
@@ -2961,7 +3047,8 @@ def test_omp_task_external_constant(fortran_reader, fortran_writer, tmpdir):
     enddo
     !$omp end task'''
     assert correct in fortran_writer(psyir)
-    assert Compile(tmpdir).string_compiles(fortran_writer(tree))
+    # Cannot test compilation of this test as it uses an external module.
+    # assert Compile(tmpdir).string_compiles(fortran_writer(tree))
 
 
 # TODO #2052 This test is expected to fail as we can't yet handle
```

Comment:
I missed this before but please could you move this whole comment into a `reason` argument for the xfail:

    @pytest.mark.xfail(reason="TODO #2052 ...")

Reply:
Done - thats a neat thing you can do.